pkg/apis/core/validation/validation.go

```diff
@@ -1833,10 +1835,18 @@ func ValidatePersistentVolumeClaimSpec(spec *core.PersistentVolumeClaimSpec, fld
 	} else if spec.DataSource != nil {
 		if len(spec.DataSource.Name) == 0 {
 			allErrs = append(allErrs, field.Required(fldPath.Child("dataSource", "name"), ""))
-		} else if !supportedDataSourceKinds.Has(string(spec.DataSource.Kind)) {
-			allErrs = append(allErrs, field.NotSupported(fldPath.Child("dataSource"), spec.DataSource.Kind, supportedDataSourceKinds.List()))
-		} else if !supportedDataSourceAPIGroups.Has(string(spec.DataSource.APIGroup)) {
-			allErrs = append(allErrs, field.NotSupported(fldPath.Child("dataSource"), spec.DataSource.APIGroup, supportedDataSourceAPIGroups.List()))
+		}
+
+		groupKind := schema.GroupKind{Group: "", Kind: spec.DataSource.Kind}
+		if spec.DataSource.APIGroup != nil {
+			groupKind.Group = string(*spec.DataSource.APIGroup)
+		}
+		groupKindList := make([]string, 0, len(supportedDataSourceAPIGroupKinds))
+		for grp := range supportedDataSourceAPIGroupKinds {
+			groupKindList = append(groupKindList, fmt.Sprintf("%#v", grp))
```

Comment:
grp.String(), right?

Reply:
If I use grp.String(), I'll get "String: VolumeSnapshot.snapshot.storage.k8s.io".

If I use fmt.Sprintf("%#v", grp), I'll get the following:
schema.GroupKind{Group:"snapshot.storage.k8s.io", Kind:"VolumeSnapshot"}

I think the second one look better?

Reply:
schema.GroupKind isn't meaningful to an API consumer. the more compact form seems better

Reply:
Ok, I'll change to grp.String()